tokio/src/fs/file.rs

```diff
@@ -394,6 +394,53 @@ impl File {
         Ok(File::from_std(std_file))
     }
 
+    /// Destructures `File` into a [`std::fs::File`][std].
```

Comment:
Would you be able to add a short paragraph under this explaining that the function is async to allow for any in-flight operations to complete and to use `try_into_std` to attempt the conversion immediately.